### PR TITLE
surf-1542-refactor-hx-file-input

### PR DESCRIPTION
**Pull request by cathysiller**
JIRA: surf-1542

### LGTM's
- [x] Dev LGTM
- [x] Zoom LGTM

### CODE REVIEW

**Comment by HelixUI** [comment]
Deploy preview for *helix-ui* processing.

Building with commit bd2d5e0e0979bc9b770cbe4c7852a1c084285b7e

https://app.netlify.com/sites/helix-ui/deploys/5cf54cb9a3d29c0008499379

File: docs/components/file/file-input-demo.js [hunk 1]
```diff
@@ -40,13 +41,21 @@ import Util from '../../_util';
...
                     return Util.snippet(`
-                        <hx-file-input
-                          ${this.attrClass}
-                          ${this.attrIcon}  
-                          ${this.attrLabel}
-                        >
-                          <input ${this.isMultiple ? 'multiple ' : ''}type="file" />
-                        </hx-file-input>
+                        <hx-file-control>
+                            <input
```
Comment (CITguy) [inline docs/components/file/file-input-demo.js]: Reduce indentation of lines 45-57 by 2 spaces

File: docs/components/file/index.html [hunk 2]
```diff
@@ -42,11 +42,17 @@ <h2 id="drop-zone">Drop Zone</h2>
           <hx-drop-fence>
             <p>
               Drop files here or
-              <hx-file-input
-                class="hxTertiary"
-                label="browse your files">
-                <input type="file" />
-              </hx-file-input>.
+              <hx-file-control>
+                  <input
```
Comment (CITguy) [inline docs/components/file/index.html]: Reduce indentation of lines 46-54 by 2 spaces

File: docs/components/file/file-input-demo.js [hunk 3]
```diff
@@ -2,11 +2,11 @@ import Util from '../../_util';
...
     const ICONS = [ 'upload', 'paperclip' ];
-    
+
     const VARIANTS = [
-        { label: 'Primary', val: 'hxPrimary' },
-        { label: 'Secondary', val: '', default: true },
-        { label: 'Tertiary', val: 'hxTertiary' },
+        { label: 'Primary', val: 'hxBtn hxPrimary' },
+        { label: 'Secondary', val: 'hxBtn', default: true },
+        { label: 'Tertiary', val: 'hxBtn hxTertiary' },
```
Comment (CITguy) [inline docs/components/file/file-input-demo.js]: Need to look into reducing repetitive `.hxBtn` class from refactored implementation, before the `v0.17.0` release.

File: src/helix-ui/elements/HXFileControlElement.js [hunk 4]
```diff
@@ -0,0 +1,26 @@
+import { HXFormControlElement } from './HXFormControlElement';
+
+/**
+ * Defines behavior for the `<hx-file-control>` element.
+ *
+ * @extends HXFormControlElement
+ * @hideconstructor
+ * @since 0.16.0
```
Comment (CITguy) [inline src/helix-ui/elements/HXFileControlElement.js]: Remove `@since`

File: src/helix-ui/elements/HXFileInputElement.less [hunk 5]
```diff
@@ -1,59 +1,25 @@
...
-    #Button.secondary();
-    #Button.medium();
-    width: 100%;
-
-    #hxLabel {
-      margin: 0;
-    }
-  }
+#hxFileInput {
+  width: 100%;
```
Comment (CITguy) [inline src/helix-ui/elements/HXFileInputElement.less]: Do we need this style? `<div>`'s are typically full width, by default.

File: src/helix-ui/styles/components/files/hx-file-control.less [hunk 6]
```diff
@@ -0,0 +1,83 @@
...
+  }
+
+  > label {
+    display: inline-block;
+    padding: 0;
+  }
+
+  > input[type="file"] {
+    #Mix.hideInput();
+    &:focus ~ label > hx-file-input {
```
Comment (CITguy) [inline src/helix-ui/styles/components/files/hx-file-control.less]: Remove this block of styles. It's already defined below.

File: src/helix-ui/styles/components/files/hx-file-control.less [hunk 7]
```diff
@@ -0,0 +1,83 @@
...
+      background-color: transparent;
+      border-color: transparent;
+      color: @cyan-700;
+    }
+  }
+}
+
+/* ========== DISABLED ========== */
+hx-file-control > input[type="file"]:disabled {
+  color: @gray-500;
```
Comment (CITguy) [inline src/helix-ui/styles/components/files/hx-file-control.less]: Is this needed?

File: src/helix-ui/elements/HXFileInputElement.less (hunk 5 above)
Comment (cathysiller) [reply src/helix-ui/elements/HXFileInputElement.less]: looks like not needed to removed

File: docs/components/file/file-input-demo.js [hunk 8]
```diff
@@ -31,22 +32,23 @@ import Util from '../../_util';
...
-                          ${this.attrIcon}  
-                          ${this.attrLabel}
-                        >
-                          <input ${this.isMultiple ? 'multiple ' : ''}type="file" />
-                        </hx-file-input>
+                        <hx-file-control>
+                          <input
+                            ${this.isDisabled ? 'disabled' : ''}
+                            ${this.isMultiple ? 'multiple ' : ''}
+                            type="file"
```
[the remark this answers is not in the document]
Comment (cathysiller) [reply docs/components/file/file-input-demo.js]: do we want to do this in this p/r or no?

File: docs/components/file/file-input-demo.js (hunk 3 above)
Comment (cathysiller) [reply docs/components/file/file-input-demo.js]: do we want to do this in this p/r or no?

File: src/helix-ui/styles/components/files/hx-file-control.less (hunk 7 above)
Comment (cathysiller) [reply src/helix-ui/styles/components/files/hx-file-control.less]: not needed so removed

File: docs/components/file/file-input-demo.js (hunk 3 above)
Comment (CITguy) [reply docs/components/file/file-input-demo.js]: not right now, but we do need a fast follow

File: src/helix-ui/styles/components/files/hx-file-control.less [hunk 9]
```diff
@@ -0,0 +1,78 @@
...
+  }
+
+  > label {
+    display: inline-block;
+    padding: 0;
+  }
+
+  > input[type="file"] {
+    #Mix.hideInput();
+  }
```
Comment (CITguy) [inline src/helix-ui/styles/components/files/hx-file-control.less]: Need to add the following CSS to remove the "dot" that appears in Edge and IE...

```less
hx-file-control > input[type="file"] {
  background: none;
}
hx-file-control > input[type="file"]::-ms-browse,
hx-file-control > input[type="file"]::-ms-value { 
  visibility: hidden; 
}
```

**Pull request merged**